Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -276,13 +276,13 @@ private static bool IsIonNamedProperty(PropertyInfo property)
         }
 
         /// <summary>
-        /// We only serde public, internal, and protected internal properties.
+        /// We only serde public, internal, and protected internal properties or properties with IonPropertyName annotation.
         /// </summary>
         private static bool HasValidAccessModifier(PropertyInfo propertyInfo)
         {
             var methodInfo = propertyInfo.GetGetMethod(true);
 
-            return methodInfo is not null && (methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly);
+            return methodInfo != null && ((methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly)) || propertyInfo.GetCustomAttribute(typeof(IonPropertyName)) != null;
```

Comment:
Correct me if im wrong, but I think the boolean logic here is slightly wrong... or at least it's written in a way that's nebulous?

There's two sets, one redundant, of braces around the methodInfo checks, and right now it's a triple composite boolean of `if methodInfo is not null AND ((methodInfo matches conditions)) OR propertyInfo has annotation`. This might work properly but it isn't the clearest. Let's make it `(if methodInfo is not null AND (methodInfo matches conditions)) OR propertyInfo has annotation`

Reply:
Thanks for pointing that out.

The logic should be: **If the property has a getter AND (property has public, internal, or protected internal access modifiers OR property has annotation)**.

Updated to:
```
return methodInfo != null && (methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly
    || propertyInfo.GetCustomAttribute(typeof(IonPropertyName)) != null);
```